pkg/iacinventory/github_repo.go

```diff
@@ -0,0 +1,184 @@
+
+	// CloudformationDirs are directories that contain cloudformation files
+	CloudformationDirs map[string]bool // DEBT: lazy map implementation
+
+	// dir is where we've unarchived the tarball for analysis.
+	dir string
+}
+
+// GetTerraformDirs implements WalkFunc to search for directories that contain Terraform files.
+func (g *GithubRepo) GetTerraformDirs(path string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
+	// if it is anything other than a regular file, return
+	if !info.Mode().IsRegular() {
+		return nil
+	}
+	// If the file ends with TF, the parent directory contains terraform files
+	if strings.HasSuffix(info.Name(), ".tf") {
+		if g.TerraformDirs == nil {
+			g.TerraformDirs = make(map[string]bool)
+		}
+		g.TerraformDirs[strings.TrimPrefix(filepath.Dir(filepath.Clean(path)), g.dir+"/")] = true
+	}
+	return nil
+}
+
+// GetCloudFormationDirs implements WalkFunc to search for CloudFormation files in a repository.
+func (g *GithubRepo) GetCloudFormationDirs(path string, info os.FileInfo, err error) error {
+	if err != nil {
+		return err
+	}
+	// if it is anything other than a regular file, return
+	if !info.Mode().IsRegular() {
+		return nil
+	}
+	// Cloudformation files do not have a unique extension, and are *typically*
+	// ".yaml" or ".json" by convention. However, sometimes organizations use
+	// Jinja, Go, or some other utility to template their Cloudformation.
+	//
+	// If the file has a possible extension and contains the string 'AWSTemplateFormatVersion',
+	// then it is *most likely* a CF file.
+
+	const maxSizeForCloudFormationConsideration int64 = 5 << (10 * 2) // 5MB, which is VERY large for json/yaml data
+	if info.Size() > maxSizeForCloudFormationConsideration {
+		// Exit early - file disqualified due to size.
+		return nil
+	}
+
+	if strings.HasSuffix(info.Name(), ".json") ||
+		strings.HasSuffix(info.Name(), ".yaml") ||
+		strings.HasSuffix(info.Name(), ".yml") ||
+		strings.HasSuffix(info.Name(), ".template") {
+		f, err := os.Open(filepath.Clean(path))
+		if err != nil {
+			return fmt.Errorf("error opening file during CloudFormation analysis: %w", err)
+		}
+		scanner := bufio.NewScanner(f)
+		for scanner.Scan() {
+			if bytes.Contains(scanner.Bytes(), []byte("AWSTemplateFormatVersion")) {
+				if g.CloudformationDirs == nil {
+					g.CloudformationDirs = make(map[string]bool)
+				}
+				g.CloudformationDirs[strings.TrimPrefix(filepath.Dir(filepath.Clean(path)), g.dir+"/")] = true
+			}
+		}
+		f.Close()
+	}
+	return nil
+}
+
+// GetMaster fetches and extracts the tarball of the master branch.
+func (g *GithubRepo) GetMaster(ctx context.Context, c *http.Client, username string, oauthToken string) error {
+	if username == "" || oauthToken == "" {
+		return fmt.Errorf("error fetching repository: credentials are not set")
+	}
+	req, _ := http.NewRequestWithContext(ctx, "GET", "https://api.github.com/repos/"+g.FullName+"/tarball/master", nil)
```

Comment:
master branch defaults to main for new repos now I believe.  If you use the google github api library it's easy to find the default branch.

Reply:
Yep, leaving this one for once I figure out how to download the repo using `go-github`.

Reply:
https://pkg.go.dev/github.com/google/go-github/v32/github#RepositoriesService.GetArchiveLink

Reply:
Solved this by just omitting the ref from the download URL. This results in downloading the default branch.